ikarus/utils/resultEvaluator.hh

```diff
@@ -0,0 +1,132 @@
+// SPDX-FileCopyrightText: 2022 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#pragma once
+
+#include "eigenDuneTransformations.hh"
+#include <dune/vtk/vtkwriter.hh>
+#include <ikarus/finiteElements/feRequirements.hh>
+
+namespace Ikarus {
+
+  enum class ResultEvaluatorComponent {
+    normalStress,
+    shearStress,
+    vonMises,
+    principalStress,
+    user_function
+  };
+  /**
+   * \brief Wrapper to evaluate results for a vtkwriter.
+   * Usage with dune-vtk : vtkWriter.addPointData(Dune::Vtk::Function<GridView>( std::make_shared<StressEvaluator2D<GridView, LinearElasticType, StressEvaluatorComponents::normalStress>>(gridView, &fes, D_Glob, lambdaLoad)));
+   * or with Dune vtkWriter.addVertexData(std::make_shared<StressEvaluator2D<GridView, LinearElasticType, StressEvaluatorComponents::normalStress>>(gridView, &fes, D_Glob, lambdaLoad));
+   */
+  template <class GridView, typename ElementType, ResultEvaluatorComponent comps, Ikarus::ResultType ResType = Ikarus::ResultType::cauchyStress, int user_ncomps = 3> requires (GridView::dimension == 2)
```

Comment:
I think it is better to pass the result type as a simple argument to the constructor and not as a template.
The calculateAt function does require it anyway as run-time argument. And recompiling is here quite useless.